### PR TITLE
ref(ts): Convert analytics utility functions to TypeScript

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/utils/analytics.tsx [hunk 1]
```diff
@@ -48,22 +48,22 @@ export const trackAnalyticsEvent = options =>
  * @param {String} options.eventKey The string key of the event to track
  * @param {...Object} options.data The parameters of the event to track
  */
-export const trackAdhocEvent = options =>
+export const trackAdhocEvent = (options: {[key: string]: any}) =>
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/analytics.tsx]: Is the value actually any? Or would `number | string | null` cover the cases we want to allow?

Comment (lynnagara) [reply src/sentry/static/sentry/app/utils/analytics.tsx]: I'm not sure if there is a restriction in what types we can send to reload. I assumed we can also send arrays / objects as well though hence `any`.

Comment (billyvg) [reply src/sentry/static/sentry/app/utils/analytics.tsx]: https://github.com/getsentry/reload/blob/133887e8856746d9e7ce2f2a1cc3205d38b412a1/reload_app/events.py

Looks like number, string, boolean

**Comment by EvanPurkhiser** [comment]
I can give a solid review on this I think, since I did a lot with this, let me read through

**Comment by lynnagara** [comment]
Aside, there seems to be a handful of similar analytics tracking code in getsentry, I wonder if it makes sense to consolidate

**Comment by EvanPurkhiser** [comment]
Actually this looks fine, I thought this was touching the the `trackAnalyticsEvent` stuff

File: src/sentry/static/sentry/app/utils/analytics.tsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/utils/analytics.tsx]: Updated

**Pull request merged**